docs/decisions/0039-template-engine.md

```diff
@@ -0,0 +1,74 @@
+---
+nav_order: 39
+parent: Decision Records
+---
+
+# Template engine
+
+## Context and Problem Statement
+
+We need to choose a template engine for [custom export filters](https://docs.jabref.org/collaborative-work/export/customexports) and [AI features](https://github.com/JabRef/jabref/pull/11884).
+
+A discussion of template engines was also in [one of the JabRef repos](https://github.com/koppor/jabref/issues/392).
+
+A discussion was raised on StackOverflow ["Velocity vs. FreeMarker vs. Thymeleaf"](https://stackoverflow.com/q/1459426/10037342).
+
+## Decision Drivers
+
+* It should support different formats, in case JabRef changes in the future. Also, AI features require just plain text manipulation.
+* It should have all necessary constructs from programming languages: loops, conditions, variables. It should support list or object (map) variables.
+* It should be embeddable and small. AI features require that templates could be generated out of simple `String` type.
```

Comment:
What does that mean? "small"? 1x1cm? Do you mean "fast"? Do you mean fast startup time?

```suggestion
* It should be fast. 
* It should be possible to provide templates out of Strings (required by the AI feature)
```